stdlib/StdlibModules

```diff
@@ -18,68 +18,71 @@
 # This file lists all standard library modules.
 # It is used in particular to know what to expunge in toplevels.
 
-P ?= stdlib__
+define add_stdlib_prefix
+ $(or $(filter stdlib, $1), $(filter camlinternal%, $1), stdlib__$1)
+endef
 
 # Modules should be listed in dependency order.
-
-STDLIB_MODULES=\
+STDLIB_MODULES=$(foreach module, \
```

Comment:
There is an unused definition in `ocamldoc/Makefile.docfiles` called `STDLIB_MODS`. I think this can be deleted from there, and `STDLIB_MODS` can be used to hold just the names, so:

```make
STDLIB_MODS=camlinternalFormatBasics stdlib pervasives seq option result \
            bool char uchar sys list …
```
(I don't believe there is any reason for each module to be on a line on its own anymore)

and have the macro fu in its own definition:

```make
STDLIB_MODULES=$(foreach module,$(call add_stdlib_prefix,$(module)))
```